### PR TITLE
feat: use ueberauth state param to always provide CSRF protection for Auth0 authentication

### LINKED ISSUE #168: CSRF error
I am getting the error below when trying to login into my application using an incognito browser:

```
%Ueberauth.Failure{
  errors: [
    %Ueberauth.Failure.Error{
      message: "Cross-Site Request Forgery attack",
      message_key: :csrf_attack
    }
  ],
  provider: :auth0,
  strategy: Ueberauth.Strategy.Auth0
}
```

The used versions are:
```
"ueberauth": {:hex, :ueberauth, "0.7.0"...
"ueberauth_auth0": {:hex, :ueberauth_auth0, "0.8.1"...
```

How ueberauth_auth0 deals with CSRF?

Also, I noticed that ueberauth implements the config `ignores_csrf_attack`. If I set it `true` in my ueberauth_auth0 fork the login works fine.

### PR DESCRIPTION
**Pull request by jmhossler**
This should fix #168. With this change, Ueberauth_auth0 will _always_ provide CSRF protection, utilizing the new ueberauth_state_param generated as of version 0.7.0 of Ueberauth.

We need this for our application, and I wanted to contribute upstream if y'all wanted the change as well. This also handles upgrading to ueberauth 0.7.0.

The one change that is breaking, in my opinion, is now the user cannot specify their own state.  The only way I could see to make this change not breaking is to keep the existing strategy the same (the only change would be to set the csrf attack protection to false), and then create a new protected strategy that is almost identical except that it doesn't allow the user to specify state, and it just uses the ueberauth_state_param by default.

I also saw an argument to forcing users to use this method, since it is more secure and would automatically protect anyone that is currently using the existing strategy without their own state param.

I'm also not sure how to update the version, but I also don't know which version the maintainer would want to bump this to. I'd say 1.0.0 myself, since the change is breaking, but I know people have a lot of opinions on versions so I'll leave that up to the maintainer.

### CODE REVIEW

File: test/strategy/auth0_test.exs [hunk 1]
```diff
@@ -58,54 +63,66 @@ defmodule Ueberauth.Strategy.Auth0Test do
...
       :get
       |> conn(
         "/auth/auth0?scope=profile%20address%20phone&audience=https%3A%2F%2Fexample-app.auth0.com%2Fmfa%2F" <>
-          "&state=obscure_custom_value&connection=facebook&unknown_param=should_be_ignored" <>
+          "&connection=facebook&unknown_param=should_be_ignored" <>
           "&prompt=login&screen_hint=signup&login_hint=user%40example.com"
       )
       |> SpecRouter.call(@router)
 
-    assert conn.resp_body == fixtures.response_advanced
```
Comment (EddieWhi) [inline test/strategy/auth0_test.exs:80]: response_advanced looks redundant now right?

Comment (jmhossler) [reply test/strategy/auth0_test.exs:80]: Yup, and response_basic! I'll remove both of those and the corresponding files to clean it up.

File: test/strategy/auth0_test.exs [hunk 2]
```diff
@@ -14,6 +14,11 @@ defmodule Ueberauth.Strategy.Auth0Test do
 
   @router SpecRouter.init([])
   @test_email "janedoe@example.com"
+  @session_options Plug.Session.init(
+                     store: Plug.Session.COOKIE,
+                     key: "_hello_key",
```
Comment (achedeuzot) [inline test/strategy/auth0_test.exs:19]: Could this be `_my_key` so we keep the convention of Elixir examples (MyApp, MyPlug, etc) ?

**Review by achedeuzot — COMMENTED** [review]
LGTM, thanks a bunch @jmhossler ❤️ 

I'm more in favor of pushing for safe & sane defaults, that includes _always_ enabling CSRF, meaning including the use of the `state` param by `ueberauth` 0.7.0.

We could provide an alternative strategy that doesn't force CSRF and allows end-users to use the `state` param as they wish, at a later stage if there is a strong use case for it (pull requests are welcome and I'll review & release them ASAP).

I'd be bumping the version to 1.0.0 as this is a breaking change of the lib and as the library itself is stable for quite some time now.

File: test/strategy/auth0_test.exs (hunk 2 above)
Comment (kevinkjt2000) [reply test/strategy/auth0_test.exs:19]:
```suggestion
                     key: "_my_key",
```

**Review by achedeuzot — APPROVED** [review]

**Comment by jmhossler** [comment]
Thanks for the quick review! I really appreciate it!

**Pull request merged**